mirrord-schema.json

```diff
@@ -1058,14 +1058,7 @@
     "TargetFileConfig": {
       "anyOf": [
         {
-          "anyOf": [
-            {
-              "$ref": "#/definitions/Target"
```

Comment:
Didn't meant to _remove_ this as a usable schema value, so need to adjust

Reply:
Ah, The problem as I see it is our use of `#[serde(deserialize_with = "string_or_struct_option")]` so schemars doesn't quite have the information that it can be a string and when providing the `#[schemars(with = "String")]` it overrides the generation process completely. 

Luckily it seems that schemars [supports passing a function](https://github.com/GREsau/schemars/blob/master/schemars_derive/src/schema_exprs.rs#L96) to create the schema and it seems simple as `Fn(&mut SchemaGenerator) -> Schema` with [`SubschemaValidation `](https://docs.rs/schemars/latest/schemars/schema/struct.SubschemaValidation.html) as a nice helper to create the correct any-of (also relevant for line [31](https://github.com/metalbear-co/mirrord/pull/2310/files#diff-c4893d2d826b5d74ff657b6e3c3f26710530785e179b4ac46e2fbb915d0cbe3aL31))

Reply:
Thanks for pointing that out, I had tried a few things with changing the Enum but didn't love how it turned out. Let me give that a shot instead!

Reply:
@DmitryDodzin I'm definitely happier with this new solution, thanks for pointing it out! The string itself is still unvalidated but it feels like a good jumping off point for the future.